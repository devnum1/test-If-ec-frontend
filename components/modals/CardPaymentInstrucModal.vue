// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="card-payment-instruc-modal" @click="closeModal()">
    <div class="modal__overlay">
      <div class="modal__container card_payment">
        <div class="modal__close-button">
          <div class="icon-delete" @click="closeModal()" />
        </div>
        <div class="modal__body">
          <div class="modal__title">
            VISA・Master・JCB・Diners{{ $t('card.such_as') }}
          </div>
          <div class="modal__card-instruc">
            <div class="image-name">
              {{ $t('card.back_face') }}
            </div>
            <div class="image--atm">
              <img src="~/assets/images/credit.png" />
            </div>
            <div class="image__description">
              {{ $t('card.security_code_instruc_3') }}
            </div>
          </div>
          <div class="modal__title">
            {{ $t('card.american_express') }}
          </div>
          <div class="modal__card-instruc">
            <div class="image-name">
              {{ $t('card.front_face') }}
            </div>
            <div class="image--credit">
              <img src="~/assets/images/AE.png" />
            </div>
            <div class="image__description">
              {{ $t('card.security_code_instruc_4') }}
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
export default {
  data() {
    return {
      display: true,
    };
  },
  methods: {
    closeModal() {
      this.display = false;
      this.$emit('input', this.display);
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/modal';
</style>
