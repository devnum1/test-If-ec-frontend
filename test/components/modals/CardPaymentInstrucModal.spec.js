import { shallowMount, config } from '@vue/test-utils';
import CardPaymentInstrucModal from '@/components/modals/CardPaymentInstrucModal.vue';

config.mocks.$t = (key) => key;

describe('CardPaymentInstrucModal', () => {
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(CardPaymentInstrucModal);
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Click close icon to call closeModal() method', () => {
    const closeBtn = wrapper.find('.icon-delete');
    wrapper.vm.closeModal = jest.fn();

    closeBtn.trigger('click');
    expect(wrapper.vm.closeModal).toHaveBeenCalled();
  });

  test('check closeModal() emits and sets value', () => {
    const closeBtn = wrapper.find('.icon-delete');

    closeBtn.trigger('click');

    expect(wrapper.emitted().input.length).toBe(2);
  });
});
