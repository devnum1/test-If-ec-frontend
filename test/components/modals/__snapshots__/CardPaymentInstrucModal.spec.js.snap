// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPaymentInstrucModal match Snapshot 1`] = `
"<div class=\\"card-payment-instruc-modal\\">
  <div class=\\"modal__overlay\\">
    <div class=\\"modal__container card_payment\\">
      <div class=\\"modal__close-button\\">
        <div class=\\"icon-delete\\"></div>
      </div>
      <div class=\\"modal__body\\">
        <div class=\\"modal__title\\">
          VISA・Master・JCB・Dinerscard.such_as
        </div>
        <div class=\\"modal__card-instruc\\">
          <div class=\\"image-name\\">
            card.back_face
          </div>
          <div class=\\"image--atm\\"><img src=\\"~/assets/images/credit.png\\"></div>
          <div class=\\"image__description\\">
            card.security_code_instruc_3
          </div>
        </div>
        <div class=\\"modal__title\\">
          card.american_express
        </div>
        <div class=\\"modal__card-instruc\\">
          <div class=\\"image-name\\">
            card.front_face
          </div>
          <div class=\\"image--credit\\"><img src=\\"~/assets/images/AE.png\\"></div>
          <div class=\\"image__description\\">
            card.security_code_instruc_4
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
